### PR TITLE
Changes after review

### PR DESCRIPTION
**Pull request by lahiru823**

### CODE REVIEW

File: components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java [hunk 1]
```diff
@@ -5988,19 +5962,14 @@ private long getApplicationAccessTokenValidityPeriod() {
...
-		Connection connection = null;
-
-		try {
-
-			connection = APIMgtDBUtil.getConnection();
-
-			boolean hasValidSubscription = false;
-
-			Subscription subscription = getIndividualSubscription(consumer, appContext, appVersion, connection);
+    public APIKeyValidationInfoDTO getApplicationData(String appContext, String appVersion, String consumer,
```
Comment (nuwand) [inline components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java:5965]: Better to have method comments in public methods which are being consumed by other components.

Comment (lahiru823) [inline components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java:5965]: fixed with:
https://github.com/lahiru823/carbon-appmgt/commit/59e0b9351417c75d9aedad43a222ec781cf79156

File: components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java [hunk 2]
```diff
@@ -4342,28 +4339,26 @@ public void addURLTemplates(int apiId, WebApp api, Connection connection) throws
...
 
 		try {
 			prepStmt = connection.prepareStatement(query);
 			Iterator<URITemplate> uriTemplateIterator = api.getUriTemplates().iterator();
-			URITemplate uriTemplate;
-			//JSONArray policyPartialIdList = (JSONArray) JSONValue.parse(api.getPolicyPartials());
+
 			while (uriTemplateIterator.hasNext()) {
-				uriTemplate = uriTemplateIterator.next();
+                URITemplate uriTemplate = uriTemplateIterator.next();
```
Comment (rushmin) [inline components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java:4355]: Formatting issue.

File: components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java [hunk 3]
```diff
@@ -6904,39 +6854,43 @@ public static Integer savePolicyGroup(String policyGroupName, String throttlingT
 			if (rs.next()) {
 				policyGroupId = Integer.parseInt(rs.getString(1));
 			}
-			// save partials mapped to policy group
-			if (objPartialMappings.length > 0) {
-				savePolicyPartialMappings(policyGroupId, objPartialMappings, conn);
-			}
+            // save partials mapped to policy group
+            if (objPartialMappings != null) {
```
Comment (rushmin) [inline components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java:6858]: Combine the two if statements for better readability.

File: components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java [hunk 4]
```diff
@@ -6904,39 +6854,43 @@ public static Integer savePolicyGroup(String policyGroupName, String throttlingT
...
-						.append(" ,isAnonymousAllowed:").append(isAnonymousAllowed)
-						.append(", Partial Mappings:").append(objPartialMappings).append(")");
-				log.debug("Record saved successfully." + strDataContext.toString());
+                strDataContext.append("(policyGroupName:").append(policyGroupName)
+                        .append(", throttlingTier:").append(throttlingTier)
+                        .append(", userRoles:").append(userRoles)
+                        .append(", isAnonymousAllowed:").append(isAnonymousAllowed)
+                        .append(", Partial Mappings:").append(objPartialMappings)
+                        .append(")");
+                log.debug("Record saved successfully." + strDataContext.toString());
```
Comment (rushmin) [inline components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java:6873]: Surround the debug log with a isDebugEnabled check.

File: components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java [hunk 5]
```diff
@@ -6984,19 +6936,23 @@ public static void updatePolicyGroup(String policyGroupName, String throttlingTi
 			//delete partials mapped to group id
 			deletePolicyPartialMappings(policyGroupId, conn);
 
-			//insert new partial mappings
-			if (objPartialMappings.length > 0) {
-				savePolicyPartialMappings(policyGroupId, objPartialMappings, conn);
-			}
+            //insert new partial mappings
+            if (objPartialMappings != null) {
```
Comment (rushmin) [inline components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java:6940]: Combine the two if statements for better readability.

File: components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java [hunk 6]
```diff
@@ -7174,14 +7130,14 @@ public void deletePolicyGroup(String applicationId, String policyGroupId) throws
...
-                    log.error("Rollback while deleting the policy group", e);
+                    log.error("Rollback while deleting the policy group : " + policyGroupId, e);
                 }
             }
-			String strDataContext =
-					"(applicationId:" + applicationId + ", policyGroupId:" + policyGroupId + ")";
-			handleException("Error while executing the query to delete XACML policies : " + query + " : " +
-					strDataContext + " : " + e.getMessage(), e);
-		} finally {
+            String strDataContext =
```
Comment (rushmin) [inline components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java:7136]: Unnecessary line break.

File: components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java (hunk 2 above)
Comment (lahiru823) [inline components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java:4355]: This issue is already fixed with a subsequent commit

File: components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java (hunk 3 above)
Comment (lahiru823) [inline components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java:6858]: commit# https://github.com/lahiru823/carbon-appmgt/commit/760441199b692e59793f74ba05f5a6f1a88e3d3d

File: components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java (hunk 5 above)
Comment (lahiru823) [inline components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java:6940]: commit # https://github.com/lahiru823/carbon-appmgt/commit/760441199b692e59793f74ba05f5a6f1a88e3d3d

File: components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java (hunk 6 above)
Comment (lahiru823) [inline components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java:7136]: This line break automatically get applied when the code is formatted with wso2 formatter

File: components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java (hunk 4 above)
Comment (lahiru823) [inline components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java:6873]: isDebugEnabled check is already there. please refer line# 6865

File: components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java [hunk 7]
```diff
@@ -4411,43 +4406,34 @@ public void updateURLTemplates(WebApp api, Connection connection) throws
...
 			prepStmt = connection.prepareStatement(query);
 			prepStmt.setString(1, apiContext);
 			prepStmt.setString(2, version);
-
 			rs = prepStmt.executeQuery();
 
-			URITemplate uriTemplate;
 			while (rs.next()) {
-				uriTemplate = new URITemplate();
+                URITemplate uriTemplate = new URITemplate();
```
Comment (dinusha-dilrukshi) [inline components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java:4424]: Formatting issue.

File: components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java [hunk 8]
```diff
@@ -4411,43 +4406,34 @@ public void updateURLTemplates(WebApp api, Connection connection) throws
...
 				uriTemplate.setSkipThrottling(rs.getBoolean("SKIP_THROTTLING"));
 				uriTemplate.setUserRoles(rs.getString("USER_ROLES"));
-
-
 				uriTemplates.add(uriTemplate);
 			}
 		} catch (SQLException e) {
-			handleException("Error while fetching all URL Templates", e);
-		} finally {
+            handleException("Error while fetching URL Templates. Api Context - "
```
Comment (dinusha-dilrukshi) [inline components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java:4434]: Formatting issue.

File: components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java [hunk 9]
```diff
@@ -4500,9 +4485,8 @@ public void updateURLTemplates(WebApp api, Connection connection) throws
 
 			resultSet = preparedStatement.executeQuery();
 
-			XACMLPolicyTemplateContext context;
 			while (resultSet.next()) {
-				context = new XACMLPolicyTemplateContext();
+                XACMLPolicyTemplateContext context = new XACMLPolicyTemplateContext();
```
Comment (dinusha-dilrukshi) [inline components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java:4489]: Formatting issue.

File: components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java [hunk 10]
```diff
@@ -4915,22 +4897,21 @@ public String createJWTTokenString(String context, String version, String subscr
 		return calleeToken;
 	}
 
-	public static List<URLMapping> getURITemplatesPerAPIAsString(APIIdentifier identifier)
-			throws
-			AppManagementException {
-		Connection conn = null;
+    public static List<URLMapping> getURITemplatesPerAPIAsString(APIIdentifier identifier)
+            throws AppManagementException {
+        Connection conn = null;
```
Comment (dinusha-dilrukshi) [inline components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java:4902]: Formatting issue.

File: components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java [hunk 11]
```diff
@@ -5124,8 +5104,8 @@ public int addComment(APIIdentifier identifier, String commentText, String user)
 			try {
 				connection.rollback();
 			} catch (SQLException e1) {
-				log.error("Failed to retrieve comments ", e);
-			}
+                log.error("Failed to retrieve comments for Application - " + identifier.getApiName(), e);
```
Comment (dinusha-dilrukshi) [inline components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java:5107]: Formatting issue.

File: components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java [hunk 12]
```diff
@@ -6017,11 +5986,11 @@ public APIKeyValidationInfoDTO getApplicationData(String appContext, String appV
 			// If there is no either an individual subscription or an enterprise subscription, don't authorize;
 			APIKeyValidationInfoDTO info = new APIKeyValidationInfoDTO();
 
-			if(!hasValidSubscription){
-				info.setValidationStatus(900908);
-				info.setAuthorized(false);
-				return info;
-			}
+            if (!hasValidSubscription) {
```
Comment (dinusha-dilrukshi) [inline components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java:5989]: Formatting issue.

File: components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java [hunk 13]
```diff
@@ -6663,19 +6628,15 @@ public static void saveStoreHits(String webAppUUID, String userId,
...
-    public static void insertUpdateStoreHits(String webAppUUID, String userId,
+    /**
+     * Insert the store wise hits
+     *
+     * @param webAppUUID Application UUID
+     * @param userId     User Id
+     * @param maxCount   Maximum hit count to be inserted
+     * @param conn       DB connection
+     */
+    private static void insertUpdateStoreHits(String webAppUUID, String userId,
```
Comment (dinusha-dilrukshi) [inline components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java:6639]: No need of using Integer here. Use int instead.

File: components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java [hunk 14]
```diff
@@ -6877,20 +6830,17 @@ public static JSONArray getAppsByHitCount(String userId,
...
-	public static Integer savePolicyGroup(String policyGroupName, String throttlingTier,
-										  String userRoles, String isAnonymousAllowed,
-										  Object[] objPartialMappings, String policyGroupDesc)
-			throws AppManagementException {
-		PreparedStatement ps = null;
-		Connection conn = null;
-		ResultSet rs = null;
-		String query = "";
-		int policyGroupId = -1;
+    public static Integer savePolicyGroup(String policyGroupName, String throttlingTier,
```
Comment (dinusha-dilrukshi) [inline components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java:6833]: Same here . Use int.

File: components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java (hunk 7 above)
Comment (lahiru823) [inline components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java:4424]: fixed with commit# https://github.com/wso2/carbon-appmgt/commit/6a0ebefd9ec0aecc3e251c3d9730c9407fa0a88f

File: components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java (hunk 8 above)
Comment (lahiru823) [inline components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java:4434]: fixed with commit# https://github.com/wso2/carbon-appmgt/commit/6a0ebefd9ec0aecc3e251c3d9730c9407fa0a88f

File: components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java (hunk 9 above)
Comment (lahiru823) [inline components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java:4489]: fixed with commit# https://github.com/wso2/carbon-appmgt/commit/6a0ebefd9ec0aecc3e251c3d9730c9407fa0a88f

File: components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java (hunk 10 above)
Comment (lahiru823) [inline components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java:4902]: fixed with commit# https://github.com/wso2/carbon-appmgt/commit/6a0ebefd9ec0aecc3e251c3d9730c9407fa0a88f

File: components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java (hunk 11 above)
Comment (lahiru823) [inline components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java:5107]: fixed with commit# https://github.com/wso2/carbon-appmgt/commit/6a0ebefd9ec0aecc3e251c3d9730c9407fa0a88f

File: components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java (hunk 12 above)
Comment (lahiru823) [inline components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java:5989]: fixed with commit# https://github.com/wso2/carbon-appmgt/commit/6a0ebefd9ec0aecc3e251c3d9730c9407fa0a88f

File: components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java (hunk 13 above)
Comment (lahiru823) [inline components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java:6639]: fixed with commit# https://github.com/wso2/carbon-appmgt/commit/111d3edde154b20054de1e9ac89f15385e57ddc9

File: components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java (hunk 14 above)
Comment (lahiru823) [inline components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java:6833]: fixed with commit# https://github.com/wso2/carbon-appmgt/commit/46dea8baf6d599e1aa1c0637ce186d42c48fa0e4

**Pull request merged**